src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -284,52 +300,88 @@ protected override void StopProcessing()
         #region ConnectionTest
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
-        {
+        {    
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpTestStatus testResult = new TcpTestStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpTestStatus.TcpConnectionTestResult.New
+                );
+                
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
+                    TcpClient client = new TcpClient();
+                    
+                    Stopwatch stopwatch = new Stopwatch();
+                    stopwatch.Start();
+
+                    Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
+                    Task timeoutTask = Task.Delay(new TimeSpan(0, 0, TimeoutSeconds));
```

Comment:
We could use Task.Run() with cancellation token.

Reply:
I've changed this a fair bit, here's a preview:

```csharp
if (client.ConnectAsync(targetAddress, TcpPort).Wait(TimeoutSeconds * 1000))
{
    testResult.Connected = true;
    testResult.Latency = stopwatch.ElapsedMilliseconds;
}
```

This makes things a bit more straightforward & readable. We still have the option to implement a cancellation token. How would you add it in this scenario? 

I'm not able to find many examples in the code that are very applicable. 

Thanks!

Reply:
Added `CancellationToken`, thanks!